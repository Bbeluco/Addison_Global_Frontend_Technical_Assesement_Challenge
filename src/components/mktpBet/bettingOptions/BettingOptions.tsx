import React from "react";
import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
import { BettingOptionsType } from "../../../types/bettingOptionsType";
import ButtonBetOption from "./buttonBetOption/ButtonBetOption";

function BettingOptions({ 
        name, 
        selections, 
        pushSelectedOptionInfo, 
        betOption, 
        eventIndex, 
        marketIndex }: BettingOptionsType): React.JSX.Element {
    return (
        <View style={styles.viewArea}>
            <Text style={styles.titleBetName}>{name}</Text>
            <View style={styles.bets}>
            {selections.map((selection, selectionIndex) => (
                <ButtonBetOption 
                    pushSelectedOptionInfo={pushSelectedOptionInfo}
                    betOption={betOption}
                    name={selection['name']}
                    price={selection['price']}
                    eventIndex={eventIndex}
                    marketIndex={marketIndex}
                    selectionIndex={selectionIndex}
                />
            ))}
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    viewArea: {
        borderWidth: 1,
        borderColor: 'black', 
    },
    bets: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    titleBetName: {
        padding: 7
    }
})

export default BettingOptions;