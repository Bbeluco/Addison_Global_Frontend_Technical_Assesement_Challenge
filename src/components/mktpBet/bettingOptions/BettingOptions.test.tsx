import BettingOptions from "./BettingOptions"
import { 
    marketNameMock, 
    selectionsMock, 
    pushSelectedOptionInfoMock,
    betOptionMock,
    eventIndexMock,
    marketIndexMock
} from "../../../mocks/mockPropsComponents";
import renderer from 'react-test-renderer';

describe('<BettingOptions />', () => {
    it('should render the component correctly', () => {
        const component = renderer.create(
            <BettingOptions
                name={marketNameMock} 
                selections={selectionsMock}
                pushSelectedOptionInfo={pushSelectedOptionInfoMock.add}
                betOption={betOptionMock}
                eventIndex={eventIndexMock}
                marketIndex={marketIndexMock}
            />
        ).toJSON();
        expect(component).toMatchSnapshot();
    })
})