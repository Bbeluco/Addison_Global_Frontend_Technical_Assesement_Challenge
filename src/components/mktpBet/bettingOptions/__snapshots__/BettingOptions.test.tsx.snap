// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BettingOptions /> should render the component correctly 1`] = `
<View
  data-testID="betting_options"
  style={
    {
      "borderColor": "black",
      "borderWidth": 1,
    }
  }
>
  <Text
    style={
      {
        "padding": 7,
      }
    }
  >
    Market name
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "green",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "margin": 5,
            "opacity": 1,
            "padding": 5,
            "width": 100,
          }
        }
        testID="button_Real Madrid_betOption"
      >
        <Text>
          Real Madrid
        </Text>
        <Text>
          1.23
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "margin": 5,
            "opacity": 1,
            "padding": 5,
            "width": 100,
          }
        }
        testID="button_Barcelona_betOption"
      >
        <Text>
          Barcelona
        </Text>
        <Text>
          2.05
        </Text>
      </View>
    </View>
  </View>
</View>
`;
